src/MSBuild/Resources/Strings.resx

```diff
@@ -1641,17 +1641,24 @@
       {4}: duration in seconds with 1 decimal point
     </comment>
   </data>
-  <data name="TestSummary" xml:space="preserve">
-    <value>Test run {0}. Total: {1} Failed: {2} Passed: {3} Skipped: {4}, Duration: {5}s</value>
-    <comment>{0} string, localized result e.g. Failed surrounded by ANSI colors.
-{1-5} whole number</comment>
+  <data name="TestSummary_BannerAndTotal" xml:space="preserve">
+    <value>Test summary: total: {0}</value>
+    <comment>{0} whole number</comment>
   </data>
-  <!-- **** TerminalLogger strings end **** -->
-    <!--
-        The command line message bucket is: MSB1001 - MSB1999
-
-        Next error code should be MSB1069.
-
-        Don't forget to update this comment after using the new code.
-  -->
-</root>
+  <data name="TestSummary_Duration" xml:space="preserve">
+    <value>duration: {0}s</value>
+    <comment>{0} whole number</comment>
+  </data>
+  <data name="TestSummary_Failed" xml:space="preserve">
+    <value>failed: {0}</value>
+    <comment>{0} whole number</comment>
+  </data>
+  <data name="TestSummary_Skipped" xml:space="preserve">
+    <value>skipped: {0}</value>
+    <comment>{0} whole number</comment>
+  </data>
+  <data name="TestSummary_Succeeded" xml:space="preserve">
+    <value>succeeded: {0}</value>
+    <comment>{0} whole number</comment>
+  </data>
```

Comment:
I was talking to some other devs/managers, and the guidance they found was to not try to chop up the overall message. Instead, you should have N copies of effectively the same message and use code conditions to choose which resource to use. This is a pain, but I think we should do that.

Reply:
I need this chopped up so I can color it using `AnsiCodes`.  What is the motivation to not having it in multiple pieces?

I can duplicate the message and embed the ANSI codes into it, but then the coloring info will be scattered over resource files. Is it worth it? We are building the message similarly in vstest, and we don't have complaints about localization of that message.

Reply:
My understanding is that we need not to chop the messages because of the localization. In different languages words' order is very different and it is safer to at least have the full sentences, or better the whole message as one entry.

Reply:
My understanding is that we need not to chop the messages because of the localization. In different languages words' order is very different and it is safer to at least have the full sentences, or better the whole message as one entry. Or else some localizations would look weird.

Reply:
In this particular example though, when you need to color the message, it might be worth it. What do you think, @baronfel?